### PR TITLE
Semester filtering

### LINKED ISSUE #164: Filter semesters by faculty (and semesters)
Single-faculty appliacations, like Fittable, need to operate only with semesters for one faculty. There should be a query parameter to get semesters only for one faculty.

### PR DESCRIPTION
**Pull request by flexik**
I've implemented semester filtering by faculty id. This PR resolves #164.

### CODE REVIEW

File: lib/sirius_api/semester_filter.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module SiriusApi
+  class SemesterFilter
+
+    def filter(dataset, params)
+      faculty_id = params[:faculty]
+      dataset.then_if(faculty_id) { |d| d.where(faculty: faculty_id) }
```
Comment (jnv) [inline lib/sirius_api/semester_filter.rb]: We have `then_if` included in other module, but you should require it in the file explicitly.

**Comment by jirutka** [comment]
@flexik Please fix the issue reported by @jnv and :shipit:.

File: lib/sirius_api/semester_filter.rb (hunk 1 above)
Comment (flexik) [inline lib/sirius_api/semester_filter.rb]: Fixed.

**Comment by flexik** [comment]
These changes are now deployed on staging.

**Pull request merged**